### PR TITLE
Feat/dashboard

### PR DESCRIPTION
**Pull request by Debarshi95**
PR to add Dashboard Page and related components.

Preview : https://625467b364ba7603643c7e6c--debarshib-flixnotes.netlify.app/
Login Creds:
Email: testuser@test.com
Password: 1234567

### CODE REVIEW

File: src/components/common/PrivateRoute/PrivateRoute.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { useAuth } from 'providers/AuthProvider/AuthProvider';
+import { Navigate } from 'react-router-dom';
+
+const PrivateRoute = ({ children }) => {
+  const { user } = useAuth();
+  if (user) {
+    return children;
+  }
+  return <Navigate to="/signin" replace />;
```
Comment (tarunsankhla) [inline src/components/common/PrivateRoute/PrivateRoute.js:9]: Why signin and not login 
Bcoz most of the website i come across they ask for login first

File: src/components/common/Typography/Typography.css [hunk 2]
```diff
@@ -46,6 +46,12 @@
 .Typography.Typography--md {
   font-size: 1.5rem;
 }
+.Typography.Typography--xxlg {
+  font-size: 2.5rem;
+}
 .Typography.Typography--primary {
   color: var(--color-primary) !important;
```
Comment (tarunsankhla) [inline src/components/common/Typography/Typography.css]: Avoid adding important you increase the specificity i know sometimes it is quicker to just add important and get things done quick, i also do it but try increasing the specificity

File: src/components/dashboard/LabelContainer/LabelContainer.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { MdClear } from 'react-icons/md';
+import Typography from 'components/common/Typography/Typography';
+import React from 'react';
+import './LabelContainer.css';
+
+const LabelContainer = ({ labels, handleLabelDelete }) => {
+  return (
+    <div className="LabelContainer__root">
+      {labels.map((label, idx) => (
```
Comment (tarunsankhla) [inline src/components/dashboard/LabelContainer/LabelContainer.js:9]: Check for labels when it is null and handle it accordingly

File: src/components/dashboard/NoteCard/NoteCard.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
+/* eslint-disable react/no-danger */
+import { Typography, LabelContainer } from 'components';
+import toast from 'react-hot-toast';
+import sanitizeHtml from 'sanitize-html';
```
Comment (tarunsankhla) [inline src/components/dashboard/NoteCard/NoteCard.js:4]: Learnt something new

**Review by tarunsankhla — APPROVED** [review]
Overall everything looks great just minor comments

**Review by m26unkwn — APPROVED** [review]
Code is clean and readable 

you can check the comments by @tarunsankhla  and then its good to go.

File: src/components/common/PrivateRoute/PrivateRoute.js (hunk 1 above)
Comment (Debarshi95) [reply src/components/common/PrivateRoute/PrivateRoute.js:9]: Signin/Login both are synonyms for each other

File: src/components/dashboard/LabelContainer/LabelContainer.js (hunk 3 above)
Comment (Debarshi95) [reply src/components/dashboard/LabelContainer/LabelContainer.js:9]: It has been already checked in the Parent component. This component only renders when parent component data has label in it.

**Pull request merged**